src/main/java/io/vertx/kafka/admin/impl/KafkaAdminClientImpl.java

```diff
@@ -500,28 +532,36 @@ public void describeLogDirs(List<Integer> brokerIds, Handler<AsyncResult<Map<Int
   public Future<Map<Integer, List<LogDirInfo>>> describeLogDirs(List<Integer> brokerIds) {
     ContextInternal ctx = (ContextInternal) vertx.getOrCreateContext();
     Promise<Map<Integer, List<LogDirInfo>>> promise = ctx.promise();
-    DescribeLogDirsResult describeLogDirsResult = this.adminClient.describeLogDirs(brokerIds);
-    describeLogDirsResult.all().whenComplete((t, ex) -> {
-      if (ex == null) {
-        Map<Integer, List<LogDirInfo>> brokerInfos = new HashMap<>();
-        for (Map.Entry<Integer, Map<String, DescribeLogDirsResponse.LogDirInfo>> brokerEntry : t.entrySet()) {
-          List<LogDirInfo> logDirInfos = new ArrayList<>();
-          for (Map.Entry<String, DescribeLogDirsResponse.LogDirInfo> logDirEntry : brokerEntry.getValue().entrySet()) {
-            LogDirInfo logDirInfo = new LogDirInfo();
-            logDirInfo
-              .setPath(logDirEntry.getKey())
-              .setReplicaInfos(logDirEntry.getValue().replicaInfos.entrySet().stream()
-                .map(v -> new TopicPartitionReplicaInfo(v.getKey(), Helper.from(v.getValue())))
-                .collect(Collectors.toList()));
-            logDirInfos.add(logDirInfo);
+    describeLogDirsInternal(brokerIds, promise);
+    return promise.future();
+  }
+
+  private void describeLogDirsInternal(List<Integer> brokerIds, Promise<Map<Integer, List<LogDirInfo>>> promise) {
+    try {
+      DescribeLogDirsResult describeLogDirsResult = this.adminClient.describeLogDirs(brokerIds);
+      describeLogDirsResult.all().whenComplete((t, ex) -> {
```

Comment:
Might be `allDescriptions` instead of `all` cause it will be deprecated since 2.7?

Reply:
Is it exists in 2.6?
https://kafka.apache.org/26/javadoc/org/apache/kafka/clients/admin/DescribeLogDirsResult.html